pkg/store/prometheus.go

```diff
@@ -176,6 +182,23 @@ func (p *PrometheusStore) Series(r *storepb.SeriesRequest, s storepb.Store_Serie
 	return nil
 }
 
+// XoR encoding supports a max size of 2^16 - 1 samples, so we need
+// to chunk all samples into groups of no more than 2^16 - 1
+func chunkSamples(series prompb.TimeSeries) [][]prompb.Sample {
```

Comment:
This name could be misleading as proper chunking should be with max 120 samples (: 

I think we should name it bit differently for reader to explicitly state it is fine to do biggestPossible chunks in our query case. Alternatively name it like this but  add `maxSamples int64` argument.

Also we could consider, as we have lot's of raw samples here to decrease mem consumption for sidecar, to reuse `e.Samples` instead of recreating multiple of arrays of `prombp.Sample`. Each prompb.Samples array with lenght 2^16 is 1MB worth of mem. Maybe not much to optimize this not sure. Depends on how many samples at the end you have. Plus with lazy GC and multiple series it might be quite large number. 

Up to you, we can start with something readable like this and move for microoptimizations later on or move to slicing and operate on indexes only e.g:
```
var aggr []storepb.AggrChunk
var arr []prompb.Sample = e.Samples // arr will share same underlying array
for len(arr) > 0 {
    len := len(arr)-1
    if i > math.MaxUint16  {
       len = math.MaxUint16
   }

    enc, cb, err := p.encodeChunk(arr[:len])
    if err != nil {
 	return status.Error(codes.Unknown, err.Error())
    }
    aggr = append(aggr, storepb.AggrChunk{
	MinTime: int64(arr[0].Timestamp),
	MaxTime: int64(sampleChunk[len-1].Timestamp),
	Raw:     &storepb.Chunk{Type: enc, Data: cb},
   })
   arr = arr[len:]
}
```

What do you think?